### PR TITLE
samples: update documentation

### LINKED ISSUE #1218: Documentation and implementation not in sync
# Bug Report

## Describe the Bug
There are situations in documentations where the explanation does not match the actual implementation.
Examples: 
In https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/tree/main/samples/03-configuration#2-access-the-config-value it's explained that one can express dependencies by overriding `requires()` from `ServiceExtension`, however in [ServiceExtension.java](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/system/ServiceExtension.java) as well as in the upstream [SystemExtension.java](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/system/SystemExtension.java) there's no `requires()` method.

Additionally, in the same file the property `dataspaceconnector.fs.config` is mentioned, which was probably renamed to `edc.fs.config`

### Expected Behavior
Current state of readme markdown files match the current state of the code

### Observed Behavior
Current state of readme markdown files does not match the current state of the code

## Steps to Reproduce
N/A

## Context Information
N/A

## Detailed Description
N/A

## Possible Implementation
adapt the readme files

### PR DESCRIPTION
**Pull request by ndr-brt**
## What this PR changes/adds

Update sample-03 documentation

## Why it does that

It was out of date.

## Further notes

-

## Linked Issue(s)

Closes #1218 

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

**Review by ronjaquensel — COMMENTED** [review]
I think it would make sense to either add a few sentences about the token-based API authentication or remove it from the build file. Currently it's there, but is never mentioned in the `README`. Therefore, when someone runs this sample and tries to call the health endpoint again to see the new log prefix, they'll get a 403.

**Review by florianrusch-zf — APPROVED** [review]

**Comment by ndr-brt** [comment]
@ronjaquensel yeah, you're right, removed it from the build file and from the readme

**Review by ronjaquensel — APPROVED** [review]

**Comment by florianrusch-zf** [comment]
> @ronjaquensel yeah, you're right, removed it from the build file and from the readme

The other samples that build on this one also use token based. Here you should possibly also take this out.

**Comment by ndr-brt** [comment]
@florianrusch-zf I don't know, in sample-04.0 the "auth key" stuff is explained in the README, maybe removing it will remove value from the sample?

**Comment by florianrusch-zf** [comment]
@ndr-brt Ahhhh sorry, of course! Then everything is fine for me so far. Thank you! :-)

File: samples/03-configuration/README.md [hunk 1]
```diff
@@ -90,19 +91,15 @@ Modify the code from the `HealthEndpointExtension.java` as shown below (use the
 of course):
 
 ```java
+@Requires(WebService.class)
```
Comment (bscholtes1A) [inline samples/03-configuration/README.md:94]: That's the old way to do it, no ?

Instead it should be:
```
@Inject
private WebService webService;
```

Comment (ndr-brt) [reply samples/03-configuration/README.md]: Don't know if that can be called "old" but yes, `@Inject` is more used nowadays, replaced it also on sample-02.

**Pull request merged**